server/resources/resources.go

```diff
@@ -152,3 +158,10 @@ func GetMyPort() (int32, error) {
 	}
 	return int32(i), nil
 }
+
+func GetZone() (string, error) {
+	if *zoneOverride != "" {
+		return *zoneOverride, nil
+	}
+	return metadata.Zone()
```

Comment:
You can check `metadata.OnGCE()` before calling `metadata.Zone()` to avoid trying to contact the metadata server if the app is not running on GCE. `metadata.OnGCE()` does a cached one time check to detect whether the binary is running on GCE.

Reply:
Oh, I guess auto-merge doesn't care if you leave a comment welp.